modules/videoModule/index.js

```diff
@@ -0,0 +1,243 @@
+import { videoImpressionVerifierFactory } from './videoImpressionVerifier.js';
+
+const videoKey = 'video';
+
+const allVideoEvents = Object.keys(videoEvents).map(eventKey => videoEvents[eventKey]);
+events.addEvents(allVideoEvents.concat([AUCTION_AD_LOAD_ATTEMPT, AUCTION_AD_LOAD_ABORT, BID_IMPRESSION, BID_ERROR]).map(getExternalVideoEventName));
+
+/**
+ * This module adds User Video support to prebid.js
+ * @module modules/videoModule
+ */
+export function PbVideo(videoCore_, getConfig_, pbGlobal_, pbEvents_, videoEvents_, gamAdServerFactory_, videoImpressionVerifierFactory_) {
+  const videoCore = videoCore_;
+  const getConfig = getConfig_;
+  const pbGlobal = pbGlobal_;
+  const requestBids = pbGlobal.requestBids;
+  const pbEvents = pbEvents_;
+  const videoEvents = videoEvents_;
+  const gamAdServerFactory = gamAdServerFactory_;
+  let gamSubmodule;
+  let mainContentDivId;
+  let contentEnrichmentEnabled = true;
+  const videoImpressionVerifierFactory = videoImpressionVerifierFactory_;
+  let videoImpressionVerifier;
+
+  function init() {
+    const cache = getConfig('cache');
+    videoImpressionVerifier = videoImpressionVerifierFactory(!!cache);
+    getConfig(videoKey, ({ video }) => {
+      video.providers.forEach(provider => {
+        videoCore.registerProvider(provider);
+        videoCore.onEvents(videoEvents, (type, payload) => {
+          pbEvents.emit(getExternalVideoEventName(type), payload);
+        }, provider.divId);
+
+        const adServerConfig = provider.adServer;
+        if (!gamSubmodule && adServerConfig) {
+          gamSubmodule = gamAdServerFactory();
+        }
+      });
+      contentEnrichmentEnabled = video.contentEnrichmentEnabled !== false;
+      mainContentDivId = contentEnrichmentEnabled ? video.mainContentDivId : null;
+    });
+
+    requestBids.before(beforeBidsRequested, 40);
+
+    pbEvents.on(CONSTANTS.EVENTS.BID_ADJUSTMENT, function (bid) {
+      videoImpressionVerifier.trackBid(bid);
+    });
+
+    pbEvents.on(getExternalVideoEventName(AD_IMPRESSION), function (payload) {
+      triggerVideoBidEvent(BID_IMPRESSION, payload);
+    });
+
+    pbEvents.on(getExternalVideoEventName(AD_ERROR), function (payload) {
+      triggerVideoBidEvent(BID_ERROR, payload);
+    });
+  }
+
+  function renderBid(divId, bid, options = {}) {
+    const adUrl = bid.vastUrl;
+    options.adXml = bid.vastXml;
+    options.winner = bid.bidder;
+    loadAdTag(adUrl, divId, options);
+  }
+
+  function getOrtbVideo(divId) {
+    return videoCore.getOrtbVideo(divId);
+  }
+
+  function getOrtbContent(divId) {
+    return videoCore.getOrtbContent(divId);
+  }
+
+  return { init, renderBid, getOrtbVideo, getOrtbContent };
+
+  function beforeBidsRequested(nextFn, bidRequest) {
+    enrichAuction(bidRequest);
+
+    const bidsBackHandler = bidRequest.bidsBackHandler;
+    if (!bidsBackHandler || typeof bidsBackHandler !== 'function') {
+      pbEvents.on(CONSTANTS.EVENTS.AUCTION_END, auctionEnd);
```

Comment:
I don't see where this listener is detached - If I run two auctions - the first with `bidsBackHandler`, the second without, I think this would still trigger on the second.